my_compassion/controllers/my2_letters.py

```diff
@@ -0,0 +1,171 @@
+##############################################################################
+#
+#    Copyright (C) 2025 Compassion CH (http://www.compassion.ch)
+#    @author: Noé Berdoz <nberdoz@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo.http import request
+from odoo import http
+
+
+class MyCompassionCorrespondenceController(http.Controller):
+
+    @http.route('/my2/children/<int:child_id>/letters', type="http", auth="user",
+                website=True)
+    def my2_render_child_letters_page(self, child_id, **kwargs):
+        partner = request.env.user.partner_id
+        children_sponsored_by_partner = partner.sponsorship_ids.child_id
+
+        letters = request.env['correspondence'].search(
+            [
+                ("partner_id", "=", partner.id)
```

Comment:
Some sponsors have access to the sponsorships of their family and they are not directly the partner of the letter. This will need to be properly fetched.